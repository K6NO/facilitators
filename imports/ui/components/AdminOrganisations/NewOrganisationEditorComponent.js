/* eslint-disable max-len, no-return-assign */
import React from 'react';
import { Table, Button, ButtonGroup, Row, Col } from 'reactstrap';
import { Bert } from 'meteor/themeteorchef:bert';
// import './NewOrganisationEditorComponent.scss';

class NewOrganisationEditorComponent extends React.Component {
  constructor(props){
    super(props);
    this.state = {
        organisation : {
            name: '',
            billingAddress: '',
            billingCity: '',
            billingPostalCode: '',
            billingCountry: '',
            active: false,
          },
          orgActiveState: false,
    }
  }

  switchOrganisationState(value) {
    this.setState({
        ...this.state,
        orgActiveState : value,
    }); 
  }

  updateOrganisationState = (field, event) => {
    this.setState({
      organisation : {
        ...this.state.organisation,
        [field]: event.target.value,
      }
    });
  }

  saveNewOrganisation = () => {
      const { closeCallback } = this.props;
      const organisation = {
          name: this.state.organisation.name,
          billingAddress: this.state.organisation.billingAddress,
          billingCity: this.state.organisation.billingCity,
          billingPostalCode: this.state.organisation.billingPostalCode,
          billingCountry: this.state.organisation.billingCountry,
          active: this.state.organisation.active,
      }
      Meteor.call('organisations.insert', organisation, (error) => {
          if(error) {
              Bert.alert(error.reason, 'danger');
          } else {
              this.setState({
                organisation : {
                    name: '',
                    billingAddress: '',
                    billingCity: '',
                    billingPostalCode: '',
                    billingCountry: '',
                    active: false,
                  },
              });
              closeCallback();
          }
      });
  }

  render() {
    const { closeCallback } = this.props;
    return ( 
        <div className="NewOrganisationEditorComponent">
            <Row>
                <Col>
                <Table>
                    <thead>
                        <tr>
                            <th>Name</th>
                            <th>Country</th>
                            <th>City</th>
                            <th>Address</th>
                            <th>Postal code</th>
                            <th></th>
                            <th></th>
                        </tr>
                        </thead>
                        <tbody>
                            <tr>
                                <td>
                                    <input name="name" type="text" 
                                    onChange={(e) => this.updateOrganisationState('name', e)} 
                                    value={this.state.organisation.name} />
                                </td>
                                <td>
                                    <input name="country" type="text"                                         
                                    onChange={(e) => this.updateOrganisationState('billingCountry', e)} 
                                    value={this.state.organisation.billingCountry} />
                                </td>
                                <td>
                                    <input name="city" type="text" 
                                        onChange={(e) => this.updateOrganisationState('billingCity', e)} 
                                        value={this.state.organisation.billingCity} />
                                </td>
                                <td>
                                    <input name="city" type="text"                                         
                                        onChange={(e) => this.updateOrganisationState('billingAddress', e)} 
                                        value={this.state.organisation.billingAddress} />
                                </td>
                                <td>
                                    <input name="city" type="text" 
                                        onChange={(e) => this.updateOrganisationState('billingPostalCode', e)} 
                                        value={this.state.organisation.billingPostalCode} />
                                </td>
                                <td>
                                    <ButtonGroup>
                                        <Button color="primary" 
                                            onClick={() => this.switchOrganisationState(true)} 
                                            active={this.state.orgActiveState === true}>Active</Button>
                                        <Button color="primary" 
                                            onClick={() => this.switchOrganisationState(false)} 
                                            active={this.state.orgActiveState === false}>Paused</Button>
                                    </ButtonGroup>
                                </td>
                                <td>
                                    <ButtonGroup>
                                        <Button color="primary" onClick={() => this.saveNewOrganisation()}>Save</Button>
                                        <Button color="danger" onClick={closeCallback}>Cancel</Button>
                                    </ButtonGroup>
                                    
                                </td>
                            </tr>
                        </tbody>
                    </Table>
                </Col>
            </Row>
        </div>
    );
  }
}


export default NewOrganisationEditorComponent;