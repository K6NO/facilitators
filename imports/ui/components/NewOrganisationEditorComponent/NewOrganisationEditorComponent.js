/* eslint-disable max-len, no-return-assign */
import React from 'react';
import { Table, Button, ButtonGroup, Row, Col } from 'reactstrap';
import { Bert } from 'meteor/themeteorchef:bert';

class NewOrganisationEditorComponent extends React.Component {
  constructor(props){
    super(props);
    this.state = {
        organisation : {
            name: '',
            country: '',
            active: false,
          },
          orgActiveState: false,
    }
  }

  switchOrganisationState(value) {
    this.setState({
        ...this.state,
        orgActiveState : value,
    }); 
  }

  updateOrganisationState = (field, event) => {
    this.setState({
      organisation : {
        ...this.state.organisation,
        [field]: event.target.value,
      }
    });
  }

  saveNewOrganisation = () => {
      const { closeCallback } = this.props;
      const organisation = {
          name: this.state.organisation.name,
          country: this.state.organisation.country,
          active: this.state.organisation.active,
      }
      Meteor.call('organisations.insert', organisation, (error) => {
          if(error) {
              Bert.alert(error.reason, 'danger');
          } else {
              //reset the state
              this.setState({
                organisation : {
                    name: '',
                    country: '',
                    active: false,
                  },
              });
              closeCallback();
          }
      });
  }

  render() {
    const { closeCallback } = this.props;
    return ( 
        <div className="NewOrganisationEditorComponent">
            <Row>
                <Col>
                <Table>
                    <thead>
                        <tr>
                            <th>Name</th>
                            <th>Country</th>
                            <th></th>
                            <th></th>
                        </tr>
                        </thead>
                        <tbody>
                            <tr>
                                <td>
                                    <input name="name" type="text" 
                                    onChange={(e) => this.updateOrganisationState('name', e)} 
                                    value={this.state.organisation.name} />
                                </td>
                                <td>
                                    <input name="country" type="text"                                         
                                    onChange={(e) => this.updateOrganisationState('country', e)} 
                                    value={this.state.organisation.country} />
                                </td>
                                <td>
                                    <ButtonGroup>
                                        <Button color="primary" 
                                            onClick={() => this.switchOrganisationState(true)} 
                                            active={this.state.orgActiveState === true}>Active</Button>
                                        <Button color="primary" 
                                            onClick={() => this.switchOrganisationState(false)} 
                                            active={this.state.orgActiveState === false}>Inactive</Button>
                                    </ButtonGroup>
                                </td>
                                <td>
                                    <ButtonGroup>
                                        <Button color="primary" onClick={() => this.saveNewOrganisation()}>Save</Button>
                                        <Button color="danger" onClick={closeCallback}>Cancel</Button>
                                    </ButtonGroup>
                                    
                                </td>
                            </tr>
                        </tbody>
                    </Table>
                </Col>
            </Row>
        </div>
    );
  }
}


export default NewOrganisationEditorComponent;