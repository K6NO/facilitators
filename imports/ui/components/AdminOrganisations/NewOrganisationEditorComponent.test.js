import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import NewOrganisationEditorComponent from './NewOrganisationEditorComponent';

test('NewOrganisationEditorComponent renders with callback function', () => {
    const mockFunction = jest.fn();
    const component = renderer.create(
        <NewOrganisationEditorComponent 
            closeCallback={mockFunction} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});