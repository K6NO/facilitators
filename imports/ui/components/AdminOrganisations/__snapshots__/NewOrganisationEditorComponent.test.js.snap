// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewOrganisationEditorComponent renders with callback function 1`] = `
<div
  className="NewOrganisationEditorComponent"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <table
        className="table"
      >
        <thead>
          <tr>
            <th>
              Name
            </th>
            <th>
              Country
            </th>
            <th>
              City
            </th>
            <th>
              Address
            </th>
            <th>
              Postal code
            </th>
            <th />
            <th />
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              <input
                name="name"
                onChange={[Function]}
                type="text"
                value=""
              />
            </td>
            <td>
              <input
                name="country"
                onChange={[Function]}
                type="text"
                value=""
              />
            </td>
            <td>
              <input
                name="city"
                onChange={[Function]}
                type="text"
                value=""
              />
            </td>
            <td>
              <input
                name="city"
                onChange={[Function]}
                type="text"
                value=""
              />
            </td>
            <td>
              <input
                name="city"
                onChange={[Function]}
                type="text"
                value=""
              />
            </td>
            <td>
              <div
                className="btn-group"
                role="group"
              >
                <button
                  aria-label={null}
                  className="btn btn-primary"
                  onClick={[Function]}
                  type="button"
                >
                  Active
                </button>
                <button
                  aria-label={null}
                  className="btn btn-primary active"
                  onClick={[Function]}
                  type="button"
                >
                  Paused
                </button>
              </div>
            </td>
            <td>
              <div
                className="btn-group"
                role="group"
              >
                <button
                  aria-label={null}
                  className="btn btn-primary"
                  onClick={[Function]}
                  type="button"
                >
                  Save
                </button>
                <button
                  aria-label={null}
                  className="btn btn-danger"
                  onClick={[Function]}
                  type="button"
                >
                  Cancel
                </button>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
